### PR TITLE
Add moon support to Node.js

### PR DESCRIPTION
**Pull request by milesj**
This PR adds moon (https://moonrepo.dev/moon) support to Node.js projects.

Technically moon can be used for any language, but will simply start with Node.js and ensure it works.

### CODE REVIEW

**Comment by milesj** [comment]
Not sure who to ping about this.

@coffee-cup

File: docs/pages/docs/providers/node.md [hunk 1]
```diff
@@ -43,15 +44,20 @@ All dependencies found in `package.json` are installed with either NPM, Yarn, PN
 
 The build script found in `package.json` if it exists.
 
-Or, if it's an NX Monorepo (detected if `nx.json` existis), the `build` pipeline for the `NIXPACKS_NX_APP_NAME` app will be called. Otherwise, it will run build for the `default_project` in `nx.json`. The build command is `(npm|pnpm|yarn|bun) run build <NxAppName>:build:production`.
+- Or, if it's a [moon repo](https://moonrepo.dev/moon) (detected if `.moon/workspace.yml` exists), the `build` task for the `NIXPACKS_MOON_APP_NAME` will be called. The task name can be customized with `NIXPACKS_MOON_BUILD_TASK`. This will run the command `moon run <app_name>:<build_task>`.
```
Comment (coffee-cup) [inline docs/pages/docs/providers/node.md:47]: Can the Moon docs please be moved below Nx and Turbo repo

**Review by coffee-cup — APPROVED** [review]
Sorry it took so long to get reviewed. Thanks for the PR adding Moon support!

Just one small comment about the ordering in docs, but besides that it looks good

File: docs/pages/docs/providers/node.md (hunk 1 above)
Comment (milesj) [reply docs/pages/docs/providers/node.md:47]: Did it alphabetical, but sure.

**Comment by coffee-cup** [comment]
Can you merge in `main`. It should have the latest lint fixes

**Pull request merged**